lxd/storage.go

```diff
@@ -638,21 +642,83 @@ func rsyncMigrationSource(container container) (MigrationStorageSourceDriver, er
 	return rsyncStorageSourceDriver{container, snapshots}, nil
 }
 
-func rsyncMigrationSink(live bool, container container, snapshots []container, conn *websocket.Conn) error {
-	/* the first object is the actual container */
-	if err := RsyncRecv(shared.AddSlash(container.Path()), conn); err != nil {
-		return err
+func snapshotProtobufToContainerArgs(containerName string, snap *Snapshot) containerArgs {
+	config := map[string]string{}
+
+	for _, ent := range snap.LocalConfig {
+		config[ent.GetKey()] = ent.GetValue()
 	}
 
-	if len(snapshots) > 0 {
-		err := os.MkdirAll(shared.VarPath(fmt.Sprintf("snapshots/%s", container.Name())), 0700)
-		if err != nil {
-			return err
+	devices := shared.Devices{}
+	for _, ent := range snap.LocalDevices {
+		props := map[string]string{}
+		for _, prop := range ent.Config {
+			props[prop.GetKey()] = prop.GetValue()
 		}
+
+		devices[ent.GetName()] = props
 	}
 
-	for _, snap := range snapshots {
-		if err := RsyncRecv(shared.AddSlash(snap.Path()), conn); err != nil {
+	name := containerName + shared.SnapshotDelimiter + snap.GetName()
+	return containerArgs{
+		Name:         name,
+		Ctype:        cTypeSnapshot,
+		Config:       config,
```

Comment:
Oh, right, we actually do the translation when parsing containerArgs, nevermind then :)